keymaster/ta/parsel.c

```diff
@@ -145,9 +169,17 @@ static bool param_deserialize(keymaster_key_param_t *param, uint8_t **buf_ptr,
 		return false;
 	case KM_BIGNUM:
 	case KM_BYTES:
+		if (TA_is_out_of_bounds(*buf_ptr, end, sizeof(uint32_t))) {
+			EMSG("Out of input array bounds on deserialization");
+			return false;
+		}
 		TEE_MemMove(&param->key_param.blob.data_length, *buf_ptr,
 			    sizeof(uint32_t));
 		*buf_ptr += sizeof(uint32_t);
+		if (TA_is_out_of_bounds(*buf_ptr, end, sizeof(offset))) {
```

Comment:
Should we combine this check with earlier TA_is_out_of_bounds check

something like
if (TA_is_out_of_bounds(*buf_ptr, end, sizeof(uint32_t) + sizeof(offset)))

Reply:
good idea, will do